### PR TITLE
[LIBCLOUD-526] Add CloudStackProject class and tests, and ability to create node with project.

### PR DESCRIPTION
**Pull request by jdivine**
https://issues.apache.org/jira/browse/LIBCLOUD-526

### CODE REVIEW

**Comment by Kami** [comment]
@jdivine Thanks, I will do a code review later today.

/cc @runseb

File: libcloud/compute/drivers/cloudstack.py [hunk 1]
```diff
@@ -402,6 +462,26 @@ def __repr__(self):
                    self.networkofferingid, self.zoneid, self.driver.name))
 
 
+class CloudStackProject(object):
+    """
+    Class representing a CloudStack Project.
+    """
+
+    def __init__(self, displaytext, name, id,
```
Comment (Kami) [inline libcloud/compute/drivers/cloudstack.py]: For consistency, `displaytext` should be `display_text`.

Comment (Kami) [inline libcloud/compute/drivers/cloudstack.py]: I also think, arguments should be defined in a different order. So:

``` python
def __init__(self, id, name display_text, driver, extra=None)
```

File: libcloud/compute/drivers/cloudstack.py [hunk 2]
```diff
@@ -781,6 +885,34 @@ def ex_list_networks(self):
...
+
+        projects = []
+        extra_map = RESOURCE_EXTRA_ATTRIBUTES_MAP['project']
+        for proj in projs:
+            extra = self._get_extra_dict(proj, extra_map)
+
+            if 'tags' in proj:
+                extra['tags'] = self._get_resource_tags(proj['tags'])
+
+            projects.append(CloudStackProject(
```
Comment (Kami) [inline libcloud/compute/drivers/cloudstack.py:906]: Minor style thing - please prefer keyword over regular arguments. So:

``` python
CloudStackProject(id=proj['id'], ...)
```

**Comment by Kami** [comment]
Lint step appears to be failing - https://travis-ci.org/apache/libcloud/jobs/20244959

**Comment by Kami** [comment]
I've added some comments. Besides that, PR looks good to me.

**Comment by sebgoa** [comment]
there is some new indentation in the list_volumes and ex_list_port_forwarding methods that are most likely breaking pep8.

**Comment by jdivine** [comment]
OK, I've made the suggested changes.  My IDE changed some indentation which caused flake8 to fail.  "displaytext" is used on other types in this driver (e.g. CloudStackNetwork) but I have made the change to "display_text" as suggested.  Thank you.

**Comment by jdivine** [comment]
Fixed a couple of other indentation issues that cropped up under the docs phase.

**Comment by Kami** [comment]
Merged into trunk.

Thanks.

**Pull request closed without merging**